### PR TITLE
ui: upgrade tslint

### PR DESCRIPTION
**Pull request by couchand**
Unfortunately a few rules triggered that weren't before, so this also
includes a handful of fixes for those.

Release note: None

### CODE REVIEW

**Comment by couchand** [comment]
This came up in the course of some other development.  I was running into a typings problem, and upgrading will fix it.

File: Makefile [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ ui-generate: $(UI_ROOT)/distccl/bindata.go
 .PHONY: ui-lint
 ui-lint: $(YARN_INSTALLED_TARGET) $(UI_PROTOS)
 	$(NODE_RUN) -C $(UI_ROOT) $(STYLINT) -c .stylintrc styl
-	$(NODE_RUN) -C $(UI_ROOT) $(TSLINT) -c tslint.json -p tsconfig.json --type-check
+	$(NODE_RUN) -C $(UI_ROOT) $(TSLINT) -c tslint.json -p tsconfig.json
```
Comment (vilterp) [inline Makefile]: is the `--type-check` flag not needed anymore?

**Review by vilterp — COMMENTED** [review]
Don't see an upgrade in `package.json`.

**Comment by couchand** [comment]
> Don't see an upgrade in package.json.

Indeed.  Done.

> is the --type-check flag not needed anymore?

Indeed not.  If you provide a `tsconfig.json` to it, now `tslint` automatically typechecks  the code if any rules require it.

**Review by vilterp — APPROVED** [review]
Seems legit.

**Pull request merged**